cmd/gen-go-sample/inittree.go

```diff
@@ -185,71 +185,57 @@ func (t *initTree) index(k string) *initTree {
 }
 
 // Just do simple structs for now.
-// TODO(pongad): allow map and array index.
-//
-// spec = path [ '=' value ] .
-func (t *initTree) parseInit(txt string, info pbinfo.Info) error {
+// TODO(hzyi): allow map index.
+func (t *initTree) parseInit(path string, value string, info pbinfo.Info) error {
 	// The first ident is treated specially to be a member of the root.
 	// Since we know the root is a struct, a dot is the only legal token anyway,
 	// so just insert a dot here so we don't need to treat the first token specially.
-	sc, report := initScanner("." + txt)
+	sc, report := initScanner("." + path)
 
-	t, r, err := t.parsePathRest(sc, info)
+	t, _, err := t.parsePathRest(sc, info)
 	if err != nil {
 		return report(err)
 	}
-	if r != '=' {
-		return report(errors.E(nil, "expected '=', got %q", r))
+	if lv := t.leafVal; lv != "" {
+		return report(errors.E(nil, "value already set to %q", lv))
 	}
 
-	// TODO(pongad): handle resource names
-	switch r := sc.Scan(); r {
-	case scanner.Int, scanner.Float, scanner.String, scanner.Ident:
-		if lv := t.leafVal; lv != "" {
-			return report(errors.E(nil, "value already set to %q", lv))
+	if _, ok := t.typ.desc.(*descriptor.DescriptorProto); ok {
+		if value != "{}" {
```

Comment:
Personally, I don't understand why we want the value to be `"{}"`, so this code confuses me (my own lack of understanding of samplegen, slowly learning more though). A comment or putting this value in a `const` (possibly with a comment at its declaration too) would be appreciated.

Furthermore, if the value is in a `const`, it can be printed in the error message with the same quotations (using `%q`) as the "found" value. It's a super nit, but rn it would be printed as `expecting '{}', found "foo"` (or something of the like).

Reply:
Good call! Added a `const emptyObjectLiteral = "{}"` to the top of the file.